src/intrinsics/prepack/global.js

```diff
@@ -551,6 +551,58 @@ export default function(realm: Realm): void {
     })
   );
 
+  // Helper function that replaces the body of a source function with another source function body.
+  // Note that this function affects un-tracked state, so care must be taken
+  // that this helper function is executed at the right time.
+  global.$DefineOwnProperty(
+    "__replaceFunctionBody",
+    new PropertyDescriptor({
+      value: new NativeFunctionValue(
+        realm,
+        "global.__replaceFunctionBody",
+        "__replaceFunctionBody",
+        2,
+        (context, [target, source]) => {
+          if (!(target instanceof ECMAScriptSourceFunctionValue)) {
+            throw realm.createErrorThrowCompletion(
+              realm.intrinsics.TypeError,
+              "first argument is not a function with source code"
+            );
+          }
+          if (!(source instanceof ECMAScriptSourceFunctionValue)) {
+            throw realm.createErrorThrowCompletion(
+              realm.intrinsics.TypeError,
+              "second argument is not a function with source code"
+            );
+          }
+
+          // relevant properties for functionValue
```

Comment:
Is it safe in general to just update all of these properties?

Reply:
I am not entirely sure, but it's fine for my purposes...